components/serialization/default/index.mjs

```diff
@@ -1,154 +1,267 @@
-const { getPrototypeOf, apply, ownKeys } = Reflect;
-const _Error = Error;
-const _Infinity = Infinity;
-const _Symbol = Symbol;
-const { keyFor, for: symbolFor } = Symbol;
-const _WeakMap = WeakMap;
-const _Map = Map;
-const _Set = Set;
-const _String = String;
-const _undefined = undefined;
-const { prototype } = Object;
-const { toString } = prototype;
-// const { isArray } = Array;
+const {
+  Reflect: { getPrototypeOf, apply, ownKeys },
+  Error,
+  Infinity,
+  Symbol,
+  Symbol: { keyFor, for: symbolFor },
+  WeakMap,
+  Map,
+  Set,
+  String,
+  undefined,
+  Object: {
+    prototype: object_prototype,
+    prototype: { toString },
+    entries: toEntries,
+  },
+  Array: { isArray },
+  JSON: { stringify: stringifyJSON },
+} = globalThis;
 
 const noargs = [];
 
 export default (dependencies) => {
   const {
-    util: { assert, hasOwnProperty, coalesce, incrementCounter, createCounter },
-    log: { logDebug },
+    util: { incrementCounter, createCounter },
+    log: { logDebug, logGuardDebug },
   } = dependencies;
+  const toEntriesSafe = (object) => {
+    try {
+      return toEntries(object);
+    } catch (error) {
+      logDebug("Failed to called Object.entries on %o >> %e", object, error);
+      return [];
+    }
+  };
+  const getSafe = (object, key) => {
+    try {
+      return object[key];
+    } catch (error) {
+      logDebug("Failed to lookup %j on %o >> %e", key, object, error);
+      return undefined;
+    }
+  };
   const empty = symbolFor("APPMAP_EMPTY_MARKER");
-  const wellknown = new _Set(
-    ownKeys(_Symbol)
-      .map((key) => _Symbol[key])
+  const wellknown = new Set(
+    ownKeys(Symbol)
+      .map((key) => Symbol[key])
       .filter((value) => typeof value === "symbol"),
   );
-  const cache = new _Map([
-    [empty, null],
-    [null, { type: "null", print: "null" }],
-    [_undefined, { type: "undefined", print: "undefined" }],
-    [true, { type: "boolean", print: "true" }],
-    [false, { type: "boolean", print: "false" }],
-  ]);
+  const printPureFallback = (value) => apply(toString, value, noargs);
+  const printPure = (value) => {
+    if (
+      value === null ||
+      value === undefined ||
+      typeof value === "boolean" ||
+      typeof value === "number"
+    ) {
+      return String(value);
+    } else if (typeof value === "string") {
+      return stringifyJSON(value);
+    } else if (typeof value === "bigint") {
+      return `${String(value)}n`;
+    } else if (typeof value === "symbol") {
+      if (wellknown.has(value)) {
+        return `well-known ${String(value)}`;
+      } else if (keyFor(value) !== undefined) {
+        return `global ${String(value)}`;
+      } else {
+        return String(value);
+      }
+    } else {
+      return printPureFallback(value);
+    }
+  };
```

Comment:
There are two reasons why I did not use `util.inspect`:

1) It is not transparent and can mess up the normal execution flow of the observed application -- eg: it can cause arbitrary code execution.
2) It is a node library and we want to eventually support other runtimes such as the browser.